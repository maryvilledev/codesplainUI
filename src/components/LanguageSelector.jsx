import React, { PropTypes } from 'react';
import {
  MenuItem,
  SelectField,
} from 'material-ui';

const LanguageSelector = (props) => {
  const {
    disabled,
    language,
    onChange,
    style,
  } = props;

  return (
    <SelectField
      disabled={disabled}
      id="languageSelector"
      onChange={onChange}
      style={style}
      value={language}
    >
      <MenuItem value="python3" primaryText="Python 3" />
      <MenuItem value="java" primaryText="Java" disabled />
    </SelectField>
  );
};

LanguageSelector.propTypes = {
  disabled: PropTypes.bool.isRequired,
  language: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default LanguageSelector;
