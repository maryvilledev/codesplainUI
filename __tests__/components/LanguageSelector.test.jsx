import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import getMuiTheme from 'material-ui/styles/getMuiTheme';

import LanguageSelector from '../../src/components/LanguageSelector';

const defaultMockProps = {
  language: 'python3',
  onChange: jest.fn(),
  style: {},
};

describe('<LanguageSelector />', () => {
  const muiTheme = getMuiTheme();
  const shallowWithContext = (node) => shallow(node, { context: { muiTheme } });

  it('matches snapshot', () => {
    const wrapper = shallowWithContext(
      <LanguageSelector {...defaultMockProps}/>
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  describe('prop: language', () => {
    it('is forwarded to SelectField\'s value prop', () => {
      const language = 'python';
      const wrapper = shallowWithContext(
        <LanguageSelector
          {...defaultMockProps}
          language={language}
        />
      );
      const select = wrapper.find('SelectField');
      expect(select.prop('value')).toEqual(language);
    });
  });

  describe('prop: onChange', () => {
    it('is forwarded to SelectField', () => {
      const onChange = jest.fn();
      const wrapper = shallowWithContext(
        <LanguageSelector
          {...defaultMockProps}
          onChange={onChange}
        />
      );
      const select = wrapper.find('SelectField');
      expect(select.prop('onChange')).toEqual(onChange);
    });
  });
});
