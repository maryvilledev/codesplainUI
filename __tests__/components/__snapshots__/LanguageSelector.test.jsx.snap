exports[`<LanguageSelector /> matches snapshot 1`] = `
<SelectField
  autoWidth={false}
  disabled={true}
  fullWidth={false}
  id="languageSelector"
  multiple={false}
  onChange={[Function]}
  style={Object {}}
  value="python3">
  <MenuItem
    anchorOrigin={
      Object {
        "horizontal": "right",
        "vertical": "top",
      }
    }
    checked={false}
    desktop={false}
    disabled={false}
    focusState="none"
    insetChildren={false}
    primaryText="Python 3"
    value="python3" />
  <MenuItem
    anchorOrigin={
      Object {
        "horizontal": "right",
        "vertical": "top",
      }
    }
    checked={false}
    desktop={false}
    disabled={false}
    focusState="none"
    insetChildren={false}
    primaryText="Java"
    value="java" />
</SelectField>
`;
